src/modules/News/controller.ts

```diff
@@ -78,4 +78,45 @@ const getTopHeadlines = async (req: Request, res: Response) => {
   }
 };
 
-export default { getLatestNews, getTopHeadlines, getPersonalizedNews };
+const getSources = async (req: Request, res: Response) => {
+  const { language, country, category } = req.query;
+
+  try {
+    const sources = await fetchFromNewsAPI('/sources', {
+      language,
+      country,
+      category,
+    });
+    res.status(200).send(sources.sources);
+  } catch (error) {
+    res.status(500).send({ message: 'Error in getting News Sources', error });
+  }
+};
+
+const getSearchResults = async (req: Request, res: Response) => {
+  const { q, from, to, sortBy } = req.query;
+
+  try {
+    const articles = await fetchFromNewsAPI('/everything', {
+      q,
+      from,
+      to,
+      sortBy,
+    });
+
+    res.status(200).send(articles.articles);
```

Comment:
The .send(<customize response>) should be according to the requirement item 2.4: 

https://github.com/StartSteps-Digital-Education-GmbH/Course-Navigator/blob/main/Curriculum/05.%20Full-Stack%20TypeScript%20Development/NEWS_APP/02_Main_API_Refrence.md

Reply:
done!:)